### ISSUE TITLE
ENHANCEMENT: Turn Customer into a DataExtension

### ISSUE THREAD

**Issue by micmania1** [swipestripe/silverstripe-swipestripe#42]
At the moment, Customer doesn't play nicely with existing systems.

If something already extends Member, whether its the user code or another module this will cause a conflict.

It also causes issues when adding the module to an existing system (as i'm finding  out) where database records are already present for Members.

Is there a reason for it being a subclass?

**Comment by stevie-mayhew** [comment]
This was by design at the beginning, but you're right - it doesn't make sense for it to not be its own distinct class.

The changes we'd make to do this would have to be targeted at a 3.0 release, which also means we can probably look at making some other changes to the architecture of the system and add or remove a bunch of features.

I'm pretty keen to start making some big changes to the way the system works, and have a lot of stuff we use internally but haven't pushed up here yet as I'm unsure what the plan is.

@frankmullenger do we have a plan? Should we make one? I'm keen.

**Comment by micmania1** [comment]
I have quite a few ideas i'd like to put forward so please keep me up to date with any progress :) You can catch me on IRC or email me at github [.at.] michaelstrong.co.uk.

**Comment by stevie-mayhew** [comment]
@icecaster @Zauberfisch what do you think, you seem to be the most active contributors.

**Comment by frankmullenger** [comment]
Hi guys, sorry have been very busy lately. I agree and it would be great to use Member instead of Customer. It was a decision made at the beginning for various reasons but Member extension would be much more convenient.

**Comment by icecaster** [comment]
@stevie-mayhew - agree, I've always wondered why this was the case.

**Comment by Zauberfisch** [comment]
I agree. Extending `Member` in SilverStripe is painful, and not worth the trouble.

I also agree that its time for some major re factoring, I have just recently launched my first SwipeStripe site, and to be honest, I had to do quiet a lot of hacking to get it working in time.

So for a new release, I think there are a great deal of edges and ends that we need to clean up, the `Customer` is one of those.

I am also willing to put in some work to improve SwipeStripe, perhaps we should draw up a strategy of what we have in mind for SwipeStripe 3.0?

**Comment by stevie-mayhew** [comment]
Yeah we have four or five sites in production now, and 3 or 4 more in development but with a largely hacked core on them all. We also don't use the default layouts or form generation so thats another challenge :smile: 

Whats the best forum for discussion for this, should we use the wiki?

**Comment by Zauberfisch** [comment]
I don't think a wiki will work well for discussion.
I think github issues or a mailinglist will work much better.

issues are probably better, because we don't have to introduce another tool, and we can easily link from commits to issues, as well as from issue to issue (if we split the conversation into several topics, which I strongly recommend)
